File: src/sdk-reference/js/6/realtime/subscribe/snippets/document-notifications-leave-scope.js
function callback (notification) {
  console.log(notification);
  /*
  { status: 200,
    requestId: 'b70997aa-682d-4beb-8d97-4dc5a786a799',
    timestamp: 1539678495284,
    volatile: { sdkVersion: '6.0.0-beta-2' },
    index: 'nyc-open-data',
    collection: 'yellow-taxi',
    controller: 'document',
    action: 'update',
    scope: 'out',
    result: { _id: 'AWZ7_WIoJD41ulNI_b-V' },
    type: 'document' }
  */
  console.log(`Document moved ${notification.scope} from the scope`);
}

try {
  // Subscribe to notifications when document leaves the scope
  const filters = { range: { age: { lte: 20 } } };
  const options = { scope: 'out' };

  await kuzzle.realtime.subscribe('nyc-open-data', 'yellow-taxi', filters, callback, options);

  const document = { name: 'nina vkote', age: 19 };

  // The document is in the scope
  const { _id } = await kuzzle.document.create('nyc-open-data', 'yellow-taxi', null, document);

  // The document isn't in the scope anymore
  await kuzzle.document.update('nyc-open-data', 'yellow-taxi', _id, { age: 42 });
} catch (error) {
  console.log(error.message);
}

File: src/sdk-reference/java/1/realtime/subscribe/snippets/document-notifications-leave-scope.java
NotificationListener listener = new NotificationListener() {
  public void onMessage(NotificationResult notification) {
    System.out.println("Document moved " + notification.getScope() + " from the scope");
  }
};

try {
    String filters = "{ \"range\": { \"age\": { \"lte\": 20 } } }";
    RoomOptions options = new RoomOptions();
    options.setScope("out");

    // Subscribe to notifications when document leaves the scope
    kuzzle.getRealtime().subscribe(
      "nyc-open-data",
      "yellow-taxi",
      filters,
      listener,
      options
    );

    String document = "{ \"name\": \"nina vkote\", \"age\": 19 }";

    // The document is in the scope
    kuzzle.getDocument().create(
      "nyc-open-data",
      "yellow-taxi",
      "nina-vkote",
      document
    );

    // The document isn't in the scope anymore
    kuzzle.getDocument().update(
      "nyc-open-data",
      "yellow-taxi",
      "nina-vkote",
      "{ \"age\": 42 }"
    );
} catch (KuzzleException e) {
    System.err.println(e.getMessage());
}
